pkg/resourcemanager/controller/managedresource/sort.go

```diff
@@ -57,3 +58,60 @@ func (r referenceSorter) Swap(i, j int) {
 	r.keys[i], r.keys[j] = r.keys[j], r.keys[i]
 	r.refs[i], r.refs[j] = r.refs[j], r.refs[i]
 }
+
+type kindSorter struct {
+	ordering map[string]int
+	objects  []object
+}
+
+func newKindSorter(obj []object, s SortOrder) *kindSorter {
+	o := make(map[string]int, len(s))
+	for v, k := range s {
+		o[k] = v
+	}
+
+	return &kindSorter{
+		objects:  obj,
+		ordering: o,
+	}
+}
+
+func (k *kindSorter) Len() int { return len(k.objects) }
+
+func (k *kindSorter) Swap(i, j int) { k.objects[i], k.objects[j] = k.objects[j], k.objects[i] }
+
+func (k *kindSorter) Less(i, j int) bool {
+	a := k.objects[i]
+	b := k.objects[j]
+	first, aok := k.ordering[a.oldInformation.Kind]
+	second, bok := k.ordering[b.oldInformation.Kind]
+
+	if !aok && !bok {
+		// if both are unknown then sort alphabetically by kind and name
+		if a.oldInformation.Kind != b.oldInformation.Kind {
+			return a.oldInformation.Kind < b.oldInformation.Kind
+		}
+		return a.oldInformation.Name < b.oldInformation.Name
+	}
+
+	// unknown kind is last
+	if !aok {
+		return false
+	}
+	if !bok {
+		return true
+	}
+
+	// if same kind sub sort alphanumeric
+	if first == second {
+		return a.oldInformation.Name < b.oldInformation.Name
+	}
+	// sort different kinds
+	return first < second
+}
+func sortByKind(resourceObject []object) []object {
+	ordering := InstallOrder
```

Comment:
By the way, the `InstallOrder` should also contain the API groups.

Reply:
@timebertt  I was trying this. Is there any actual scenario where just kind base sorting will not work? I looked at the resources that are applied by the resource manager I didn't see where we can face issues based on kind base sorting.

Reply:
I have two things in my mind first sort based on apigroup and then by kind and apply resources in the resulted order but this can't be done.
Second, the order is the same, how is it today(in this PR)  plus also concatenate apigroup with it, but I didn't find the use of it. Where do we need it I am not able to find a scenario?

Reply:
We can't know, what resources are contained in a `ManagedResource` and whether their kinds will conflict with the kubernetes built-in kinds.
But we can also change that later on, right.